Packages/Core/Editor/Generator/Templates/PairTemplateFactory.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using System.Net;
+using UnityEngine;
+
+namespace UnityAtoms.Editor
+{
+    public static class PairTemplateFactory
+    {
+        public const string atomName = "Pair";
+
+        public static Template GetTemplate(Type type, Func<Type, Template> templateFactory)
+        {
+            var pairType = typeof(Pair<>).MakeGenericType(type);
+
+            var template = templateFactory.Invoke(pairType);
+            var typeName = type.CSharpName().Capitalize();
+            var pairTypeName = pairType.CSharpName().Capitalize();
+
+            template.className = template.className.Replace(pairTypeName, typeName) + atomName;
```

Comment:
Appending Pair to the end does not seem right to me. The result becomes for example `ByteEventInstancerPair`, but the desired result is `BytePairEventInstancer`.